framework/dynamic.go

```diff
@@ -81,65 +81,136 @@ func (frame *Framework[VertexProp, EdgeProp, MsgType]) EnactStructureChanges(g *
 			// From the gathered set of consecutive adds, apply them.
 			if len(src.OutEdges) > didxStart {
 				val := frame.AggregateRetrieve(src)
-				frame.OnEdgeAdd(g, sidx, didxStart, val)
+				msgs := frame.OnEdgeAdd(g, sidx, didxStart, val)
+				if g.Undirected || g.SendRevMsgs {
+					for eidx := didxStart; eidx < len(src.OutEdges); eidx++ {
+						edge := src.OutEdges[eidx]
+						didx := edge.Destination
+						var msg MsgType
+						if msgs != nil { // Just incase this is not needed/implemented
+							msg = msgs[eidx-didxStart]
+						}
+						// Target dest since we swap s/d for the reverse
+						// TODO: Some consideration needed for edgeproperties here... (should not be shared mem / pointers)
+						g.ReverseMsgQ[g.RawIdToThreadIdx(g.Vertices[didx].Id)] <- graph.RevMessage[MsgType, EdgeProp]{Message: msg, EdgeProperty: edge.Property, Type: graph.ADD, Didx: sidx, Sidx: didx}
```

Comment:
Why do we need to include EdgeProperty in RevMessage? Can we let the user include it as part of the Message?

Reply:
This is a tricky one. For an undirected graph, its not clear what an edge property should reference. It could be a shared property such that each side of the undirected edge points to the same property, but this would be hard to do with a distributed implementation.
For this I've opted to go with a duplication of the property, so it's passed along when building the reverse edge to be applied there. The algorithm could also optionally view the forward edge's property this way to make a decision when it is notified of the creation.

Reply:
Got it! Thanks!